### PR TITLE
Implemented smt2 option ":print-cores-full" (from CVC5)

### PR DESCRIPTION
**Pull request by Tomaqa**

### CODE REVIEW

**Comment by Tomaqa** [comment]
See the discussion in #765 on the combination with minimal unsat cores

**Comment by Tomaqa** [comment]
If ":minimal-unsat-cores" is set as well, we have to minimize all terms, not only the named ones.

**Comment by Tomaqa** [comment]
I verified that the current outputs are consistent with CVC5.

File: test/regression/base/generic/redundant_min_unsat_core2_full.smt2.expected.out [hunk 1]
```diff
@@ -0,0 +1,2 @@
+unsat
+( (and b1 b2) (xor b2 b1) )
```
Comment (blishko) [inline test/regression/base/generic/redundant_min_unsat_core2_full.smt2.expected.out:2]: There seems to be some problem here, the arguments of `xor` are swapped.
Maybe some bug in the printing?

**Review by blishko — COMMENTED** [review]
Looks good, but it seems we have some problem with handling/printing `xor`.

File: test/regression/base/generic/redundant_min_unsat_core2_full.smt2.expected.out (hunk 1 above)
Comment (Tomaqa) [reply test/regression/base/generic/redundant_min_unsat_core2_full.smt2.expected.out:2]: I noticed that but did not care too much - I thought it is related to some preprocessing.
Nonetheless, I think this should be resolved in a separate issue.
-> #776

Comment (Tomaqa) [reply test/regression/base/generic/redundant_min_unsat_core2_full.smt2.expected.out:2]: OK!

**Pull request merged**